submissions/aymkin/OOP-JS (tjsw)/index.js

```diff
@@ -0,0 +1,73 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+
+Code repository: https://github.com/aymkin/a-tiny-JS-world
+Web app: https://aymkin.github.io/a-tiny-JS-world/
+   */
+
+function Inhobitant(species, name, gender, say, friends = []) {
+	this.species = species;
+	this.name = name;
+	this.gender = gender;
+	this.friends = friends;
+	this.say = say;
+}
+
+Inhobitant.prototype.makeSound = function() {
+	return this.say;
+};
+
+Inhobitant.prototype.getFriend = function() {
+	return this.hasOwnProperty('friends') && this.friends.length > 0
+		? `I have friends: ${this.friends.join(', ')}.`
+		: `I have no friends`;
+};
+
+Inhobitant.prototype.getBio = function() {
+	return `I am ${this.species}, my name is <strong>${
+		this.name
+	}</strong>. ${this.getLimbs()}. ${this.getFriend()} \nEvery day I say: ${this.makeSound()}.`;
```

Comment:
Now, here is the problem:
Class `Inhobitant` doesn't keep any info about limbs.
So the following code
```
let a = new Inhobitant('dog','Aaa','male','Woof');
a.getBio();
```
will produce error `Uncaught TypeError: this.getLimbs is not a function`, which means the code is inconsistent.

The above means that you need to employ overloading, i.e. child classes should implement their own `.getBio` methods. To avoid code duplication child class `.getBio` can call parent's `.getBio`, add something and return a composed result.

The pattern of calling parent's method:
`ParentClass.prototype.myMethod.call(this, arg1, arg2, ..)`

Reply:
As far as I could... done), check it well, please

Reply:
Inhobitant doesn't define `.limbs` and it shouldn't. But this also means that `Inhobitant.getBio` cannot refer to an inexistent property `.limbs`.

Reply:
Please, re-read carefully the initial comment in this thread.